src/sentry/api/endpoints/organization_details.py

```diff
@@ -522,15 +522,16 @@ def put(self, request: Request, organization) -> Response:
                     result = serializer.validated_data
 
                     if "slug" in changed_data:
-                        organization_mapping_service.create(
+                        organization_mapping_service.reserve_slug_for_organization(
                             organization_id=organization.id,
                             slug=organization.slug,
                             name=organization.name,
                             idempotency_key=result.get("idempotencyKey", ""),
                             customer_id=organization.customer_id,
                             region_name=settings.SENTRY_REGION or "us",
+                            status=organization.status,
                         )
-                    elif "name" in changed_data:
+                    elif "name" in changed_data or "status" in changed_data:
                         organization_mapping_service.update(
                             organization_id=organization.id,
                             update=RpcOrganizationMappingUpdate(name=organization.name),
```

Comment:
We also need the status set here, right?

Reply:
Ahh yep! thanks 😅

Reply:
I wonder if we should just make a `classmethod` on `RpcOrganizationMappingUpdate` that takes an organization and creates a `RpcOrganizationMappingUpdate` instance as the result.  Therefore we can get one implementation that always sets all fields anyways and reduces chance of missing something?

Reply:
That's a solid suggestion. Fwiw, `organization_details` and `organization_index` are the only two files that appear to interact with these methods, but I could do this refactor now if we're afraid of this becoming more of a problem in the future